bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/FreeAtHomeSystemDiscoveryService.java

```diff
@@ -0,0 +1,135 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.freeathomesystem.internal;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.freeathomesystem.internal.datamodel.FreeAtHomeDeviceDescription;
+import org.openhab.binding.freeathomesystem.internal.handler.FreeAtHomeBridgeHandler;
+import org.openhab.binding.freeathomesystem.internal.util.FreeAtHomeHttpCommunicationException;
+import org.openhab.core.config.discovery.AbstractThingHandlerDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FreeAtHomeSystemDiscoveryService} is responsible for performing discovery of things
+ *
+ * @author Andras Uhrin - Initial contribution
+ */
+@NonNullByDefault
+public class FreeAtHomeSystemDiscoveryService extends AbstractThingHandlerDiscoveryService<FreeAtHomeBridgeHandler> {
+
+    private final Logger logger = LoggerFactory.getLogger(FreeAtHomeSystemDiscoveryService.class);
+    private @Nullable ScheduledFuture<?> backgroundDiscoveryJob = null;
+
+    private static final long BACKGROUND_DISCOVERY_DELAY = 1L;
+    private boolean isScanTerminated;
+
+    Runnable runnable = new Runnable() {
+        @Override
+        public void run() {
+            ThingUID bridgeUID = thingHandler.getThing().getUID();
+
+            List<String> deviceList;
+
+            try {
+                deviceList = thingHandler.getDeviceDeviceList();
+
+                for (int i = 0; (i < deviceList.size()) && !isScanTerminated; i++) {
+                    FreeAtHomeDeviceDescription device = thingHandler.getFreeatHomeDeviceDescription(deviceList.get(i));
+
+                    ThingUID uid = new ThingUID(FreeAtHomeSystemBindingConstants.FREEATHOMEDEVICE_TYPE_UID, bridgeUID,
+                            device.deviceId);
+                    Map<String, Object> properties = new HashMap<>(1);
+                    properties.put("deviceId", device.deviceId);
+                    properties.put("interface", device.interfaceType);
+
+                    String deviceLabel = device.deviceLabel;
+
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(uid).withLabel(deviceLabel)
```

Comment:
To avoid duplicate discovery results - assuming that `device.deviceId` is unique and constant:
```suggestion
                    DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(uid).withLabel(deviceLabel).withRepresentationProperty(device.deviceId)
```